apis/rabbitmq/v1beta1/transporturl_types.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TransportURLSpec defines the desired state of TransportURL
+type TransportURLSpec struct {
+	// +kubebuilder:validation:Required
+	// RabbitmqClusterName the name of the Rabbitmq cluster which to configure the transport URL
+	RabbitmqClusterName string `json:"rabbitmqClusterName,omitempty"`
+}
+
+// TransportURLStatus defines the observed state of TransportURL
+type TransportURLStatus struct {
+	// SecretName - name of the secret containing the rabbitmq transport URL
+	SecretName string `json:"secretName,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// TransportURL is the Schema for the transporturls API
+type TransportURL struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   TransportURLSpec   `json:"spec,omitempty"`
+	Status TransportURLStatus `json:"status,omitempty"`
+}
```

Comment:
?: I assume the `cinder-operator` would watch this CR and should update its services when it changes, but how would it know that the the `Spec` and `Status` are in sync?

Reply:
And an admission webhook that would clear that flag when the Spec is changed, right?

Reply:
Or you could hash the spec and store it in the status, and then change condition when old hash does not equal current hash

Reply:
As I originally had this the existence of the secret means it was successful. This wouldn't cover the case for updates however so we should have some conditions and use those instead to determine readyness.